src/Regression/MultipleRegression.cs

```diff
@@ -1,113 +1,137 @@
+            }
 
-    internal override (double[][] trainingInputs, double[] trainingOutputs, double[][] oosInputs, double[] oosOutputs) 
-        PrepareData(double[][] inputs, double[] outputs, int trainingSize, INormalization? normalization)
-    {
-        return normalization?.PrepareData(inputs, outputs, trainingSize) ?? NormalizationHelper.SplitData(inputs, outputs, trainingSize);
-    }
+            // Choose decomposition method based on MatrixDecomposition option and solve matrix
+            switch (RegressionOptions.MatrixDecomposition)
+            {
+                case MatrixDecomposition.Cholesky:
+                    inputMatrix.Cholesky().Solve(outputVector, result);
+                    break;
+                case MatrixDecomposition.Evd:
+                    inputMatrix.Evd().Solve(outputVector, result);
+                    break;
+                case MatrixDecomposition.GramSchmidt:
+                    inputMatrix.GramSchmidt().Solve(outputVector, result);
+                    break;
+                case MatrixDecomposition.Lu:
+                    inputMatrix.LU().Solve(outputVector, result);
+                    break;
+                case MatrixDecomposition.Qr:
+                    inputMatrix.QR().Solve(outputVector, result);
+                    break;
+                case MatrixDecomposition.Svd:
+                    inputMatrix.Svd().Solve(outputVector, result);
+                    break;
+                default:
+                    throw new ArgumentException("Invalid MatrixDecomposition option");
+            }
 
-    internal override double[] Transform(double[][] inputs)
-    {
-        var predictions = new double[inputs[0].Length];
+            // Set coefficients and YIntercept
+            Coefficients = result.ToArray();
+            YIntercept = 0;
+        }
 
-        for (var i = 0; i < inputs.Length; i++)
+        // PrepareData method
+        internal override (double[][] trainingInputs, double[] trainingOutputs, double[][] oosInputs, double[] oosOutputs) 
+            PrepareData(double[][] inputs, double[] outputs, int trainingSize, INormalization? normalization)
         {
-            for (var j = 0; j < inputs[j].Length; j++)
-            {
-                predictions[j] += YIntercept + Coefficients[i] * inputs[i][j];
-            }
+            // Prepare data based on normalization option
+            return normalization?.PrepareData(inputs, outputs, trainingSize) ?? NormalizationHelper.SplitData(inputs, outputs, trainingSize);
         }
 
-        return predictions;
+        // Transform method
+        internal override double[] Transform(double[][] inputs)
+        {
+            // Initialize predictions array and store lengths for optimization
+            var predictions = new double[inputs[0].Length];
+            var inputsLength = inputs.Length;
+            var inputs0Length = inputs[0].Length;
```

Comment:
It is a minor issue but you are referencing the same inputs[0].Length on line 121 and 123 so you should move up your variable and use that variable for line 121